ui/app/rif/pages/rns/subdomains/index.js

```diff
@@ -4,36 +4,187 @@ import PropTypes from 'prop-types';
+    table: 'n-table',
+    thead: '',
+    theadTr: '',
+    theadTh: '',
+    tbody: '',
+    tbodyTr: '',
+    tbodyTd: 'n-table-td',
+    noData: '',
+    content: 'n-table-content-channels',
+    contentActions: 'n-table-actions',
+    customButton: {
+      button: 'btn-add',
+      icon: '',
+      text: '',
+    },
+    pagination: {
+      body: 'n-table-pagination',
+      buttonBack: 'n-table-pagination-back',
+      indexes: '',
+      activePageButton: 'n-table-pagination-active',
+      inactivePageButton: 'n-table-pagination-inactive',
+      buttonNext: 'n-table-pagination-next',
+    },
+  },
+}
 
 class Subdomains extends Component {
 
   static propTypes = {
-    classes: PropTypes.any,
+    subdomain: PropTypes.object.isRequired,
+    domainName: PropTypes.string.isRequired,
+    pageName: PropTypes.string.isRequired,
+    redirectParams: PropTypes.any.isRequired,
+    selectedResolverAddress: PropTypes.string,
+    isOwner: PropTypes.bool,
+    newChainAddresses: PropTypes.array,
+    showPopup: PropTypes.func,
+    deleteSubdomain: PropTypes.func,
+    waitForListener: PropTypes.func,
+    showTransactionConfirmPage: PropTypes.func,
+    showThis: PropTypes.func,
+    showToast: PropTypes.func,
+    getSubdomains: PropTypes.func,
   }
 
   constructor (props) {
     super(props);
+    const resolvers = Object.assign([], GET_RESOLVERS());
     this.state = {
+      resolvers: resolvers,
     };
   }
 
+  openDeletePopup (subdomain) {
+    this.props.showPopup('Delete Subdomain', {
+      text: 'Are you sure you want to delete the subdomain ' + subdomain.name + '?',
+      confirmCallback: async () => {
+        const transactionListenerId = await this.props.deleteSubdomain(subdomain.domainName, subdomain.name);
+        this.props.waitForListener(transactionListenerId).then(transactionReceipt => {
+          this.props.getSubdomains(this.props.domainName)
+            .then(subdomains => {
+              this.props.showThis(
+                this.props.pageName,
+                {
+                  ...this.props.redirectParams,
+                  newSubdomains: subdomains,
+                });
+              });
+            });
+        this.props.showTransactionConfirmPage({
+          action: () => {
+            this.props.showThis(
+              this.props.pageName,
+              this.props.redirectParams);
+            this.props.showToast('Waiting for confirmation');
+          },
+        });
+      },
+      confirmButtonClass: '',
+    });
+  }
+
   render () {
+    const { subdomain, domainName, selectedResolverAddress, newChainAddresses, isOwner } = this.props;
+    const displayName = domainName + '.' + subdomain.name;
+    const { resolvers } = this.state;
     return (
-      <div>Subdomains</div>
+      <div>
+        <DomainHeader
+          domainName={displayName}
+          showOwnerIcon={isOwner}
+        >
+          {isOwner &&
+            <CustomButton
+              text={'Delete'}
```

Comment:
When you have literal values like a string you can use literal declaration like this text="Delete", it's not necessary to use variable value since you are not using a variable but static value.